### PR TITLE
RESTful API Server Integration with conditional compilation

### PR DESCRIPTION
**Pull request by NeblioTeam**
Pass REST=1 at compile time to compile in support for the RESTful API Server

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -5,6 +5,9 @@
 #include "txdb.h"
 #include "walletdb.h"
 #include "bitcoinrpc.h"
+#ifdef REST
```
Comment (TheQuantumPhysicist) [inline src/init.cpp]: This is probably too generic. Maybe we should call this NEBLIO_REST instead. Conflicts in defs are ugly.

File: src/init.cpp [hunk 2]
```diff
@@ -892,7 +895,15 @@ bool AppInit2()
     if (fServer)
         NewThread(ThreadRPCServer, NULL);
 
-    // ********************************************************* Step 12: finished
+    // ********************************************************* Step 12: start rest listenser
+
+#ifdef REST
+    uiInterface.InitMessage(_("Starting RESTful API Listener"));
+    printf("Starting RESTful API Listener\n");
+    NewThread(ThreadRESTServer, NULL);
```
Comment (TheQuantumPhysicist) [inline src/init.cpp]: I'm unable to find the function ThreadRESTServer() anywhere. Is this still not implemented? I'm guessing so.

File: src/init.cpp [hunk 3]
```diff
@@ -892,7 +895,15 @@ bool AppInit2()
...
+#endif
+
+    // ********************************************************* Step 13: finished
 
     uiInterface.InitMessage(_("Done loading"));
```
Comment (TheQuantumPhysicist) [inline src/init.cpp]: Probably this should synchronously wait for the rest server to finish loading before saying "Done loading". We can do this with a std::promise/std::future pair. Adding a TODO comment here wouldn't be the worst idea.

File: src/init.cpp (hunk 2 above)
Comment (NeblioTeam) [reply src/init.cpp]: It is implemented in nebliorest.cpp which is not yet included.

File: src/init.cpp (hunk 3 above)
Comment (NeblioTeam) [reply src/init.cpp]: Agreed. Will make the TODO for now.

File: src/init.cpp (hunk 1 above)
Comment (NeblioTeam) [reply src/init.cpp]: agreed

**Comment by NeblioTeam** [comment]
Added new commit to address comments

**Pull request merged**